### DISCUSSION TITLE
Float toolbar on remote connection window always starts expanded, connection popup also

### DISCUSSION THREAD

**Discussion by FCrane** [rustdesk/rustdesk#4424] [General]
### Bug Description

Hi! The float menu bar always starts expanded (fully visible) when a connection is started. It collapses (hides) automatically after a while, but during that time, it obscures important information on the remote desktop! Why does it always have to show expanded??? I rarely need it (if at all). With recent nightily builds, I can finally move the arrow out of the way (and it stays there), but the unneeded menu bar still blocks important information. Please at least add an option to not show the menu bar each time a new connection is established.

Also, the large connection window that pops up every time a connection is started is very annoying (but at least it is located in the corner). There should be a way of preventing that popup, too! I use RustDesk to quickly monitor a remote machine that displays a lot of info all over the desktop, so it's really annoying that some of it is always obscured by unneeded popups, menus, arrows, etc. and I have to wait to see it all.

Thanks!

### How to Reproduce

See description.

### Expected Behavior

No popups, menus, etc. showing when starting a connection.

### Operating system(s) on local side and remote side

Windows 11

### RustDesk Version(s) on local side and remote side

Latest nightly

### Screenshots

n/a

**Comment by deep-soft** [comment]
My 2 cents..
- Connection popup must be visible. Those on the remote computer must be aware...
- Float menubar, if possible, must have an option in the settings: expanded / not expanded.

**Reply by Bruce-gits** [reply]
I am using this software to monitor 25 stations for work computers, all the my sub-ordinates have drops on sales and work efficiency because they are starting to use work computers for personal since we switch to Rustdesk because each time we do an audit , they know when we access the work stations. WE DO NOT allow work computers for personal matters and hence for doing this , they get warning and second time will be fired. THERE SHOULD BE AN ADMIN OPTION TO TURN OFF NOTIFICATION when there is a connection on a password.

**Reply by rustdesk** [reply]
> * Float menubar, if possible, must have an option in the settings: expanded / not expanded.

we will add this

**Reply by deep-soft** [reply]
About Float menubar:
It's possible to have only one bar?
I mean "the handle" (which control show/hide buttons bar) and the buttons bar to be on the same row.
         [button bar]
    [handle]
to be:
    [handle][button bar]
Trying to upload as images:
this:
![image](https://github.com/rustdesk/rustdesk/assets/10506197/91a69afa-9536-4492-9f1f-6c8e55f08e5f)
change to this:
![image](https://github.com/rustdesk/rustdesk/assets/10506197/c2f74af7-f158-4013-b085-d160f53fd639)

**Reply by 21pages** [reply]
#4614

**Reply by deep-soft** [reply]
> #4614

Thank you for this, that was quick!
Testing now ...

**Comment by rustdesk** [comment]
![image](https://github.com/rustdesk/rustdesk/assets/71636191/3c0c9be3-d665-4b1c-8dcb-8928ce606e86)

**Reply by mubeenkh4u** [reply]
Is this only applicable when running a server?

**Reply by Dhoot** [reply]
can't find this setting

**Reply by User-Win7** [reply]
![IMG_20251206_234903](https://github.com/user-attachments/assets/b74441f2-5391-48f3-ba8f-58fb618bfcbd)
Ahh I see, is it this one?

**Comment by rustdesk** [comment]
It is called toolbar, https://github.com/rustdesk/rustdesk/pull/3670#issuecomment-1470438228

**Comment by FCrane** [comment]
Why can't I enable the option "Hide connection management window" remotely??? Again the same problem: when I start the RustDesk main program, the client area turns gray and I can't click anything! What's the non-sense behind this??? Do I really need to attacha monitor, mouse and keyboard to my headless server, just to be able to configure the most basic things of RustDesk??? Or do I have to install TeamViewer, Parsec or whatever just to configure RustDesk??? Come on, this is absurd... It also doesn't allow me to upgrade the remote version of RustDesk, because the installer window always goes gray when I move the mouse into it.

Please fix this EXTREMELY ANNOYING BUG! I need to be able to change ALL options remotely and I need to upgrade to the latest nightly remotely as well without having to attach a monitor and mouse every time...

Thanks!

**Comment by Bruce-gits** [comment]
FCrane, the option is there, its called 'enable remote configuration'. 
Its absurb that people dont spend time researching or reading and expect or learn the manual on a FREE open source software that developers have donated their valuable time to provide this great software.

**Comment by FCrane** [comment]
Thanks for pointing that out - but where is that "Manual"? On RustDesk.com, there seems to be nothing and on GitHub, I just found an FAQ... The program itself also has no "Help", just points to the website...

Unfortunately, this is very counter-intuitive! Dimming the whole window gray when the mouse moves inside it, looks much more like a bug than a feature! It's absurd, especially because there seem to be at least 2 ways to get around it:

1) You can still open the "Settings" by clicking on the icon in the caption. If you then don't move the mouse inside the window, everything stays "active", sou you can "Tab" your way through the settings and change them, even if you should not be able to.

2) There is a short delay when moving the mouse inside the client area, so if you are quick, you can still click items! I managed to "Unlock security settings" this way, although I shouldn't be. As soon as they are unlocked, graying stops and I can change the settings normally.

So this "graying" thing is not only absolutely counter-intuitive, but buggy and just annoying! The window should simply present a message like "Remote configuration disabled in the security options" and then, everything would be clear, intuitive and logical. Same applies for upgrading the program - there needs to be a way to do it, even when the user initially forgets to allow remote configuration (e.g. by prompting for the password again, etc.)

So I finally managed to get rid of this graying annoyance, but still the ""Hide connection management window"" is grayed out and I can not enable it. Why? I'm using "Use permanent password", but the checkbox below never gets active... Again no clue from the app why this is happening. And no manual or FAQ either...

Regards!

**Reply by rustdesk** [reply]
There is a tooltip if you hover mouse on the grayed `Hide connection management window`. As for document, it is open source, https://github.com/rustdesk/doc.rustdesk.com. If every one is waiting for us to write document, honestly, you will be disappointed.

**Comment by FCrane** [comment]
Yes, I saw the tooltip, which says "Alllow hiding only if accepting sessions via password and using permanent password" - what is this supposed to mean? The remote side is accepting sessions via password and it is using a permanent password! Still, the option is always disabled...

BTW:
First of all, a tooltip should explain what the option does, not what the requirements are for it to be accessible! Using tooltips for that is not standard UI design and should therefore be avoided.

Secondly, all mentioned requirements are met and still the option is inaccessible. Why? What's the point of the tooltip then?

As for the manual: nobody expects (or asks) you to maintain a manual - the program should be intuitive to use and configure and self-eplanatory (after all, nobody wants to read manuals anyway). I just asked for it, because "Bruce-gits" told me to RTFM instead of asking why something obviously does not work as it should. So it turns out there is no manual (and the documentation you mention is not for Windows, but for Android, Linux, etc.) 

Again: I do not mind there being no manual, but the program should be intuitive enough to not need a manual. Graying out the client area (only) when the mouse moves inside it, mentioning option requirements in tooltips which then don't work anyway, etc. is not intuitive and only leads to confusion and frustration.

I'm just trying to point out some obvious flaws to help improving this great project, so that everybody can use it and will say "wow, this simply works!" (unlike all other remote control programs out there).

Regards!

**Reply by 21pages** [reply]
Choose `Accept sessions via password`.

![1686560259623](https://github.com/rustdesk/rustdesk/assets/14891774/f4d09257-1a0b-4516-9794-ffa1064eb87c)

**Comment by FCrane** [comment]
Ahhh, ok - again very counter-intuitive and well hidden! "Accept via both" was selected, but looked like it was disabled, because it is all gray an nothing indicates, that this should be changed! See:

![image](https://github.com/rustdesk/rustdesk/assets/14123400/e7926640-4fc8-4a07-9fec-b116739b9166)

The choice of UI colors is confusing and misleading here. Also, "Both" should actually include "Password", so this is also confusing...

Anyway, this seems to work now - but I hope you see why this is "difficult" to configure...

Now just one more thing to fix: the toolbar needs to start collapsed, because i hardly ever need it! Or is there also an option somewhere for that which I wasn't able to find?

Thanks!

**Reply by deep-soft** [reply]
> Now just one more thing to fix: the toolbar needs to start collapsed, because i hardly ever need it! Or is there also an option somewhere for that which I wasn't able to find?
> 
> Thanks!

https://github.com/rustdesk/rustdesk/pull/4614
https://github.com/rustdesk/rustdesk/discussions/4424#discussioncomment-6143322

**Comment by Bruce-gits** [comment]
hi, why was the 'hide management window' option removed on the latest version? I need this to monitor a few stations computers

**Reply by rustdesk** [reply]
https://github.com/rustdesk/rustdesk/discussions/5624

**Reply by ThePapaw** [reply]
I have been looking for it as well, will it be added back?

**Reply by Mattias16-beep** [reply]
Hello. 

I was looking for this option ? Why this option has gone ? 
Is there a way to bring back this option? 

@rustdesk
